source/css/sass/_grid.scss

```diff
@@ -1,16 +1,83 @@
+@import "variables/color";
+@import "mixins/decorations";
+@import "mixins/grid";
 @import "mixins/spacing";
 
-.wrapper {
-  @include wrapper();
+.grid-item-span-full {
+  @include grid-span-full();
 }
 
-.grid {
-  @supports (display: grid) {
-    display: grid;
+.grid-item-span-main {
+  @include grid-span-main();
+}
+
+.page-grid {
+  @include grid-setup-base(map_get($grid, column-gap));
+  @include grid-setup-basic-columns(
+    map_get($grid, edge-colum-count),
+    map_get($grid, column-gap),
+    map_get($grid, max-width)
+  );
+}
+
+.page-grid-item {
+  @include grid-setup-base(map_get($grid, column-gap));
+  @include grid-setup-basic-columns(
+    map_get($grid, edge-colum-count),
+    map_get($grid, column-gap),
+    map_get($grid, max-width)
+  );
+
+  @include grid-span-full();
+
+  & > * {
+    @include grid-span-main();
+  }
+
+  & > .grid-item-span-full {
+    @include grid-span-full();
+  }
+
+  &.page-grid-item--has-keyline-start {
+    @include divider(block-start);
+  }
+
+  &.page-grid-item--has-keyline-end {
+    @include divider(block-end);
   }
-  // consider setting some sensible gap defaults here.
+
+}
+
+.grid-research-article {
+  @include grid-setup-base(map_get($grid, column-gap));
+  @include grid-span-full();
+
+  & > .grid-primary-content {
+    grid-column: primary;
+  }
+
+  & > .grid-tertiary-content {
+    grid-column: tertiary;
+  }
+
+}
+
+.grid-research-article--no-secondary-content {
```

Comment:
Could rename this to be more generic, e.g. `.grid-content--no-secondary` (similarly on [line 73](https://github.com/libero/pattern-library/pull/54/files#diff-6be14bafaef56e74ed55cc62c8776465R73)).